src/Concerns/IsDeprecated.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\dsql\Concerns;
+
+/**
+ * use to trigger deprecated warning on a class.
+ */
+trait IsDeprecated
+{
+    public function __construct()
+    {
+        @trigger_error('Use ' . parent::class . ' instead', E_USER_DEPRECATED);
```

Comment:
why @ before, will it actually emit anything?

If constructor is not used, this will not trigger... I think it is better to move it before/after the class def directly and emit these warnings always when the class is inluded.

Reply:
The @ is added by CS Fixer.

I will move it in the file